File: backend/src/main/java/com/homework/backend/job/model/Job.java
package com.homework.backend.job.model;

import com.homework.backend.enums.JobType;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table
public class Job {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	private String title;
	private String description;
	private float salary;
	private String location;
	private String postcode;
	private JobType jobType;
	
	private int user_id;
	
	public Job() {
		
	}

	public Job(
			String title, String description, float salary,
			String location, String postcode, int user_id,
			JobType jobType
			) {
			this.title = title;
			this.description = description;
			this.salary = salary;
			this.location = location;
			this.postcode = postcode;
			this.user_id = user_id;
			this.jobType = jobType;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public float getSalary() {
		return salary;
	}

	public void setSalary(float salary) {
		this.salary = salary;
	}

	public String getLocation() {
		return location;
	}

	public void setLocation(String location) {
		this.location = location;
	}

	public String getPostcode() {
		return postcode;
	}

	public void setPostcode(String postcode) {
		this.postcode = postcode;
	}

	public int getUser_id() {
		return user_id;
	}

	public void setUser_id(int user_id) {
		this.user_id = user_id;
	}

	public JobType getJobType() {
		return jobType;
	}

	public void setJobType(JobType jobType) {
		this.jobType = jobType;
	}
}

File: backend-asp-net-core/Models/Job.cs
using backend_asp_net_core.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;

namespace backend_asp_net_core.Models
{
    public class Job
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [AllowNull]
        public string Title { get; set; }
        [AllowNull]
        public string Description { get; set; }
        [AllowNull]
        public float Salary { get; set; }
        [AllowNull]
        public string Location { get; set; }
        [AllowNull]
        public string Postcode { get; set; }
        [AllowNull]
        public JobType JobType { get; set; }

        [AllowNull]
        public int User_id { get; set; }

        public Job(string title, string description, float salary, string location, string postcode, JobType jobType, int user_id)
        {
            Title = title;
            Description = description;
            Salary = salary;
            Location = location;
            Postcode = postcode;
            JobType = jobType;
            User_id = user_id;
        }
    }
}
